sdcm/sct_events/base.py

```diff
@@ -76,6 +83,16 @@ def filter_by_shard(self, shard: int) -> ContinuousRegistryFilter:
 
         return self
 
+    def filter_by_alert(self, alert: str) -> ContinuousRegistryFilter:
```

Comment:
These filter methods are so specific to Prometheus alerts, that I'm not sure if we should include them here. Maybe a cleaner solution would be to create a subclass for the `ContinuousRegistryFilter` for `PrometheusAlertEvents` (same builder pattern) with these specific methods only and add a `get_prometheus_alerts_registry_filter` to the `ContinuousEventsRegistry` class. The `get_pattern_to_event_to_func_mapping` function could then create the filter it needs based on the match found in the log line.

Reply:
Moved ContinuousEvent to it's own module